src/lib/assets/assets.test.ts

```diff
@@ -240,44 +267,63 @@ describe("Assets", () => {
   });
 
   it("should call generateHelmChart which should call createDirectoryIfNotExists twice for templates and charts", async () => {
-    const webhookGeneratorFunction = jest
-      .fn<() => Promise<V1MutatingWebhookConfiguration | V1ValidatingWebhookConfiguration | null>>()
-      .mockResolvedValue(new kind.MutatingWebhookConfiguration());
-    await assets.generateHelmChart(webhookGeneratorFunction, "/tmp");
+    const webhookGeneratorFunction = createMockWebhookGenerator();
+    const getWatcherFunction = createMockWatcher();
+    const getModuleSecretFunction = createMockModuleSecret();
+
+    await assets.generateHelmChart(
+      webhookGeneratorFunction,
+      getWatcherFunction,
+      getModuleSecretFunction,
+      "/tmp",
+    );
     expect(createDirectoryIfNotExists).toHaveBeenCalledTimes(2);
   });
 
   it("should call generateHelmChart which should write file 40 times for built Kubernetes Manifests and helm chart generation", async () => {
-    const webhookGeneratorFunction = jest
-      .fn<() => Promise<V1MutatingWebhookConfiguration | V1ValidatingWebhookConfiguration | null>>()
-      .mockResolvedValue(new kind.MutatingWebhookConfiguration());
-    await assets.generateHelmChart(webhookGeneratorFunction, "/tmp");
+    const webhookGeneratorFunction = createMockWebhookGenerator();
+    const getWatcherFunction = createMockWatcher();
+    const getModuleSecretFunction = createMockModuleSecret();
```

Comment:
We _could_ even inline these function calls as part of the "act" in this test, but that's a stylistic preference.